pom.xml

```diff
@@ -2136,6 +2136,41 @@
             </properties>
         </profile>
 
+        <profile>
+            <id>java-21</id>
+            <activation>
+                <jdk>21</jdk>
+            </activation>
+            <properties>
+                <java.version>21</java.version>
+                <maven.compiler.source></maven.compiler.source>
+                <maven.compiler.target></maven.compiler.target>
+                <maven.compiler.release>${java.version}</maven.compiler.release>
+                <maven.plugin.surefire.argLine>-XX:+IgnoreUnrecognizedVMOptions
+                    --add-opens=java.base/java.lang=ALL-UNNAMED
+                    --add-opens=java.base/java.lang.invoke=ALL-UNNAMED
+                    --add-opens=java.base/java.lang.reflect=ALL-UNNAMED
+                    --add-opens=java.base/java.io=ALL-UNNAMED
+                    --add-opens=java.base/java.net=ALL-UNNAMED
+                    --add-opens=java.base/java.nio=ALL-UNNAMED
+                    --add-opens=java.base/java.util=ALL-UNNAMED
+                    --add-opens=java.base/java.util.concurrent=ALL-UNNAMED
+                    --add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED
+                    --add-opens=java.base/sun.nio.ch=ALL-UNNAMED
+                    --add-opens=java.base/sun.nio.cs=ALL-UNNAMED
+                    --add-opens=java.base/sun.security.action=ALL-UNNAMED
+                    --add-opens=java.base/sun.security.tools.keytool=ALL-UNNAMED
+                    --add-opens=java.base/sun.security.x509=ALL-UNNAMED
+                    --add-opens=java.base/sun.util.calendar=ALL-UNNAMED
+                    -Djdk.reflect.useDirectMethodHandle=false
+                    -Dio.netty.tryReflectionSetAccessible=true</maven.plugin.surefire.argLine>
+                <!-- Current google format does not run on Java 21.
+                     Don't upgrade it in this profile because it formats code differently.
+                     Re-evaluate once support for Java 8 is dropped. -->
```

Comment:
can we recover it by override `<spotless.java.googlejavaformat.version>` in this profile?

Reply:
But if using a new version of `googleJavaFormat` for code formatting check under Java 21, one problem is that this would cause some existing code formatting checks to fail, like this:
```
[ERROR] Failed to execute goal com.diffplug.spotless:spotless-maven-plugin:2.43.0:check (default) on project kyuubi-hive-jdbc: The following files had format violations:
[ERROR]     src/main/java/org/apache/kyuubi/jdbc/KyuubiDriver.java
[ERROR]         @@ -17,6 +17,8 @@
[ERROR]          
[ERROR]          package·org.apache.kyuubi.jdbc;
[ERROR]          
[ERROR]         -/**·@deprecated·Use·`KyuubiHiveDriver`·instead.·*/
[ERROR]         +/**
[ERROR]         +·*·@deprecated·Use·`KyuubiHiveDriver`·instead.
[ERROR]         +·*/
[ERROR]          @Deprecated
[ERROR]          public·class·KyuubiDriver·extends·KyuubiHiveDriver·{}
```
If set `check` to `apply`, in which case it would work.

Reply:
If we accepts those changes, will the Java8 with old `googleJavaFormat` be happy?

Reply:
I will take come tests using both Java 8 and Java 21, and then report back here.

Reply:
I upgraded the versions of `spotless` and `googlejavaformat` in the java21 profile and manually executed `mvn spotless:apply`, some code formats were adjusted by `spotless`. 

But when I switched to Java8 with old `googlejavaformat` for checking, errors occurred. It seems that the rules of the new version of `googlejavaformat` are not compatible with the old version. Perhaps before we drop support for Java8, we can only skip spotless's check in Java21. Once we abandon support for Java8, we can upgrade the versions of spotless and googlejavaformat.

Reply:
thanks for investigating, let's leave a brief comment to explain the decision, i.e. "Skip Java formatting because Java 21 requires a new version of googlejavaformat, but the format rules conflict with the existing one"